doc/developer-documentation.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "Developer Documentation"
+date: 2024-05-10
+description: "KernelCI API/Pipeline developer manual"
+weight: 5
```

Comment:
Should this document be listed on the docs page or just available in the repo? I did not see it on https://static.staging.kernelci.org/docs/api/ although this patch has been pulled to staging branch (maybe `weight` is too low?)

Reply:
I am not sure. Maybe it's because `local-instance` page didn't have `weight` property.
I added it. There are some issues with the staging deployment atm.
I'll verify if once it's fixed.

Reply:
Due to issues in staging website deployment, I deployed it locally for verification.